### DISCUSSION TITLE
How to modify response body using middleware?

### DISCUSSION THREAD

**Discussion by oldtune** [actix/actix-web#2864] [Q&A] [answered]
Hi, i'm trying to standadize my API response to something like this
```
BaseResponse<T> where T: Serialize{
Success: bool,
Data: T,
Error: Vec<String>,
}
```

but not sure how to extract the response body and wrap it inside the `call` function
```
fn call(&self, req: ServiceRequest) -> Self::Future {
        let fut = self.service.call(req);

        Box::pin(async move {
            let mut res = fut.await?;
              //Having no clue how to unwrap the body and wrap it in BaseResponse here
            Ok(res)
        })
    }
```
I've been scratching my head on Google and StackOverflow but still can't figure out how to make it work, can you show me if it's possible?

**Comment by robjtede** [comment, marked as answer]
See https://github.com/actix/actix-web/discussions/2834 for info and implications on getting response body in middleware.
